### PR TITLE
Fix TracAlgorithm with DDPG

### PR DESCRIPTION
**Pull request by emailweixu**
Also makes it work with EntropyTargetAlgorithm

### CODE REVIEW

File: alf/algorithms/trac_algorithm.py [hunk 1]
```diff
@@ -239,16 +227,13 @@ def _update_total_dists(new_action, exp, total_dists):
...
-                new_action = policy_step.action
-            else:
-                assert (
-                    common.is_namedtuple(policy_step.info)
-                    and "action_distribution" in policy_step.info._fields
-                ), ("PolicyStep.info from ac_algorithm.predict() should be "
-                    "a namedtuple containing `action_distribution` in order to "
-                    "use TracAlgorithm.")
-                new_action = policy_step.info.action_distribution
+            assert (
```
Comment (witwolf) [inline alf/algorithms/trac_algorithm.py]: redundant assertion, already checked in `_make_policy_step`

File: alf/algorithms/ddpg_algorithm.py [hunk 2]
```diff
@@ -157,10 +159,13 @@ def _sample(a, ou):
                 tf.less(tf.random.uniform((), 0, 1),
                         epsilon_greedy), lambda: a + ou(), lambda: a)
 
-        action = tf.nest.map_structure(_sample, action, self._ou_process)
+        noisy_action = tf.nest.map_structure(_sample, action, self._ou_process)
```
Comment (witwolf) [inline alf/algorithms/ddpg_algorithm.py]: should clip noise action  to match action spec

File: alf/algorithms/trac_algorithm.py (hunk 1 above)
Comment (emailweixu) [reply alf/algorithms/trac_algorithm.py]: That only asserts for train_step() and rollout(). Here is predict().

File: alf/algorithms/ddpg_algorithm.py (hunk 2 above)
Comment (emailweixu) [reply alf/algorithms/ddpg_algorithm.py]: fixed.

**Pull request merged**